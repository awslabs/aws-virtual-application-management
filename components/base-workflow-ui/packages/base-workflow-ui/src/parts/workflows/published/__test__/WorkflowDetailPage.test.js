/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://aws.amazon.com/apache2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react';
import WorkflowDetailPage from '../WorkflowDetailPage';

jest.mock('../../../workflow-common/ProgressPlaceholder');
jest.mock('../../../workflow-common/component-states/WorkflowCommonCardState');
jest.mock('../WorkflowDetailTabs');

describe('WorkflowDetailPage', () => {
  const userDisplayName = {
    isSystem: jest.fn(),
    getDisplayName: jest.fn(() => {
      return 'test-getDisplayName';
    }),
  };
  const workflowsStore = {
    load: jest.fn(),
    getWorkflowStore: jest.fn(() => {
      return {
        ready: 'store-ready',
        load: jest.fn(),
        startHeartbeat: jest.fn(),
        workflow: {
          versionNumbers: [1],
          getVersion: jest.fn(() => {
            return {
              id: 'test-id',
              title: 'test-title',
              updatedAt: 'Feb 1, 1966 UTC',
              updatedBy: 'test-updatedBy',
              desc: 'test-desc',
            };
          }),
        },
      };
    }),
  };
  const match = { params: {} };

  beforeEach(() => {
    jest.spyOn(global.Date, 'now').mockImplementation(() => 1626365860332);
  });

  it('WorkflowDetailPage should renders correctly', () => {
    expect(
      renderer
        .create(
          <Provider workflowsStore={workflowsStore}>
            <WorkflowDetailPage.WrappedComponent
              match={match}
              workflowsStore={workflowsStore}
              userDisplayName={userDisplayName}
            />
          </Provider>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });
});
