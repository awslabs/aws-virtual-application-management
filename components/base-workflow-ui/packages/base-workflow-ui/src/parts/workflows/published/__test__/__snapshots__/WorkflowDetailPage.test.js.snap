// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowDetailPage WorkflowDetailPage should renders correctly 1`] = `
<div
  className="ui container mt3"
>
  <div
    className="ui breadcrumb block mb3"
  >
    <a
      className="section"
      onClick={[Function]}
    >
      Workflows
    </a>
    <i
      aria-hidden="true"
      className="right angle icon divider"
      onClick={[Function]}
    />
    <div
      className="active section"
      onClick={[Function]}
    />
  </div>
  <div
    className="flex mb2"
  >
    <h3
      className="ui grey header mt0 flex-auto ellipsis"
    >
      <div
        className="ui teal label ml0 mr1"
        onClick={[Function]}
      >
        Workflow
      </div>
      test-title
      <div
        className="sub header fs-9 color-grey mt1"
      >
        <div>
          updated 
          <time
            dateTime="1966-02-01T00:00:00.000Z"
            title="Feb 1, 1966 UTC"
          >
            55 years ago
          </time>
           
          <span
            className="ml1"
          >
            by 
            test-getDisplayName
          </span>
        </div>
      </div>
    </h3>
    <div
      className="ml1"
    >
      <span
        className="ellipsis pr1 fs-9 breakout color-grey"
      >
        test-id
      </span>
       
      <span
        className="bold color-grey pr2"
      >
        v
        NaN
      </span>
    </div>
  </div>
  <div
    className="mb3"
  >
    <p>
      test-desc
    </p>
  </div>
</div>
`;
